addons/project/controllers/portal.py

```diff
@@ -542,3 +544,33 @@ def portal_my_task(self, task_id, report_type=None, access_token=None, project_s
             request.session['my_tasks_history'] = task_sudo.ids
         values = self._task_get_page_view_values(task_sudo, access_token, **kw)
         return request.render("project.portal_my_task", values)
+
+    @http.route('/project_sharing/attachment/add_image', type='http', auth='user', methods=['POST'], website=True)
+    def add_image(self, name, data, res_model, res_id, access_token=None, **kwargs):
+        if res_model != 'project.task':
+            raise UserError(_("Invalid request!"))
+        try:
+            task_sudo = self._document_check_access(res_model, int(res_id), access_token=access_token)
+            if not task_sudo.with_user(request.env.uid).project_id._check_project_sharing_access():
+                raise AccessError
+        except (AccessError, MissingError) as e:
+            raise UserError(_("The document does not exist or you do not have the rights to access it."))
+
+        IrAttachment = request.env['ir.attachment']
+
+        # Avoid using sudo when not necessary: internal users can create attachments,
```

Comment:
Even if you don't have access to project.task ?

Reply:
don't `_document_check_access` and `_check_project_sharing_access` guarantee that ?